src/gluonnlp/sequence_sampler.py

```diff
@@ -717,11 +717,20 @@ def forward(self, samples, valid_length, outputs, scores, step, beam_alive_mask,
         probs = mx.npx.softmax(outputs / self._temperature)
 
         if self._sampling_topp > 0:
-            probs = mx.np.where(
-                probs > self._sampling_topp,
-                probs,
-                mx.np.zeros_like(probs)
-            )
+             sorted_probs, sorted_indices = mx.npx.topk(probs, axis=2, k=-1, ret_typ='both', is_ascend=False)
+             cumsum_probs = mx.np.cumsum(sorted_probs, axis=2)
+             masked_probs = mx.np.where(
+                 cumsum_probs > self._sampling_topp,
+                 sorted_probs,
+                 mx.np.zeros_like(probs)
+             )
+             # choose the borderline prob
+             p_prob = mx.np.min(masked_probs, axis=2, keepdims=True)
```

Comment:
Is it possible to use exactly the same implementation as https://gist.github.com/thomwolf/1a5a29f6962089e871b94cbd09daf317?

Reply:
I don't think sort is implemented yet.

Reply:
I'm referring to the part in which they choose not to mask the top-1 probability:
```
sorted_indices_to_remove[..., 1:] = sorted_indices_to_remove[..., :-1].clone()
```

Reply:
Sorry for the confusion. I see that both sort and argsort are implemented but I don't see a way to get both values and indices in one call. The usage of `topk(k=-1)` that assumes the return values to be sorted seems to be undocumented, which is a bit of a concern.